doc/en/user/source/extensions/css/workshop/linestring.rst

```diff
@@ -0,0 +1,500 @@
+
+.. figure:: img/LineStringLabel.svg
+   
+   Use of Label Property
+
+This is also our first example making use of a dynamic style (where the value of a property is defined by an attribute from your data).
+
+#. To enable LineString labeling we will need to use the key properties for both **stroke** and **label**.
+
+   Update ``line_example`` with the following:
+   
+   .. code-block:: css
+      :emphasize-lines: 2,3
+
+      * {
+        stroke: blue;
+        label: [name];
+      }
+
+#. The SLD standard documents the default label position for each kind of Geometry. For LineStrings the initial label is positioned on the midway point of the line.
+
+   .. image:: img/line_label_1.png
+
+#. We have used an expression to calculate a property value for label. The **label** property is generated dynamically from the :kbd:`name` attribute. Expressions are supplied within square brackets, making use of Constraint Query Language (CQL) syntax. 
+
+   .. code-block:: css
+      :emphasize-lines: 3
+
+      * {
+        stroke: blue;
+        label: [name];
+      }
+
+#. Additional properties can be supplied to fine-tune label presentation:
+   
+   .. code-block:: css
+      
+      * {
+        stroke: blue;
+        label: [name];
+        font-fill: black;
+        label-offset: 7px;
+      }
+
+#. The **font-fill** property is set to :kbd:`black` provides the label color.
+
+   .. code-block:: css
+      :emphasize-lines: 4
+      
+      * {
+        stroke: blue;
+        label: [name];
+        font-fill: black;
+        label-offset: 7px;
+      }
+      
+#. The **label-offset** property is used to adjust the starting position used for labeling.
+   
+   Normally the displacement offset is supplied using two numbers (allowing an x and y offset from the the midway point used for LineString labeling).
+
+   When labeling a LineString there is a special twist: by specifying a single number for **label-offset** we can ask the rendering engine to position our label a set distance away from the LineString. 
+  
+   .. code-block:: css
+      :emphasize-lines: 5
+      
+      * {
+        stroke: blue;
+        label: [name];
+        font-fill: black;
+        label-offset: 7px;
+      }
+
+#. When used in this manner the rotation of the label will be adjusted automatically to match the LineString.
+
+   .. image:: img/line_label_2.png
+
+How Labeling Works
+------------------
+
+The rendering engine collects all the generated labels during the rendering of each layer. Then, during labeling, the engine sorts through the labels performing collision avoidance (to prevent labels overlapping). Finally the rendering engine draws the labels on top of the map. Even with collision avoidance you can spot areas where labels are so closely spaced that the result is hard to read.
+
+The parameters provided by SLD are general purpose and should be compatible with any rendering engine.
```

Comment:
A nit pick, but to prevent confusion, should probably say "the SLD specification, from which our CSS implementation is derived" instead of just "SLD" Maybe I'm not far enough into this, but my thought was "but we're not working with SLD." Thought it might have been a typo.